storage/watchable_store.go

```diff
@@ -356,3 +360,25 @@ type watching struct {
 	// The chan might be shared with other watchings.
 	ch chan<- storagepb.Event
 }
+
+// safelySynced safely puts watching with key k into watchableStore's synced.
+func safelySynced(s *watchableStore, k string, wa *watching) error {
```

Comment:
What does safely means? 

Also it is better to make the func a verb. Like PutIntoSynced or AddWatching.
Moreover, why do you need to pass in the watchableStore? I think you only access the map.